[IMP] essentials: search filter and group

samueljlieber opened the pull request:
Task: [#3843408](https://www.odoo.com/web#id=3843408&cids=3&menu_id=4720&action=333&active_id=3835&model=project.task&view_type=form)

Improves the Search and Filter documentation for 17.0 with the following improvements:

- Renamed h1 to Search, filter, and group records
- Removed any pronoun use
- Updated content and screenshots for 17.0
- Expanded the Custom Filters section with details
- Added the Favorites section

samueljlieber commented:
Hi @tiku-odoo, this PR is ready for a peer review whenever you have a chance! Thank you :)

tiku-odoo reviewed (approved):
Hi @samueljlieber 

Great work on this Search/Filter PR! It's a huge improvement. 

I've made a handful of suggestions, some grammar/spelling, and then some clarifications. 

What do you think about adding some highlights to the images? If not, they work as is, just thought they might direct the "quick glance users" to the feature's location better. 

Great, work. Let me know if you have any questions. I've approved this PR. 

Thanks, Tim :+1:

samueljlieber commented on content/applications/essentials/search.rst:
```diff
@@ -1,100 +1,206 @@
-   You could *only* include records from a single salesperson on the Sales Analysis report, for
-   example *Mitchell Admin*, by selecting :guilabel:`Salesperson` as the field, :guilabel:`is equal
-   to` as the operator, and typing `Mitchell Admin` as the value.
+The :guilabel:`Add Custom Filter` modal displays the matching option, filter rule, and a toggle to
```
The term is correct here, _modal_ is the UI term for the pop-up window on the screen. However, we are still deciding on whether or not to use this terminology. So I will be changing this to "pop-up window" for now :)

samueljlieber commented:
Thank you for your helpful review @tiku-odoo, I implemented your suggestions and added an example for branch merges. Passing to @ksc-odoo for final content review 👍

ksc-odoo commented on content/applications/essentials/search.rst:
```diff
@@ -1,100 +1,214 @@
+Odoo allows for the searching, filtering, and grouping of records in a view to display only the most
+relevant records. The search bar is located at the top of the view, start typing to :ref:`search for
+values <search/values>`, or click the :guilabel:`🔽 (down arrow) icon` to access the :ref:`Filter
+<search/filters>`, :ref:`Group By <search/group>` and :ref:`Favorites <search/favorites>` drop-down
```
```suggestion
<search/filters>`, :ref:`Group By <search/group>`, and :ref:`Favorites <search/favorites>` drop-down
```

ksc-odoo commented on content/applications/essentials/search.rst:
```diff
@@ -1,100 +1,214 @@
+Search for values
+=================
+
+Use the search field to quickly look for specific values and add them as a filter. Type the value to
```
```suggestion
Use the search field to quickly look for specific values, and add them as a filter. Type the value to
```

ksc-odoo commented on content/applications/essentials/search.rst:
```diff
@@ -1,100 +1,214 @@
+=================
+
+Use the search field to quickly look for specific values and add them as a filter. Type the value to
+search for and select the desired option from the drop-down menu to apply the search filter.
```
```suggestion
search for, and select the desired option from the drop-down menu to apply the search filter.
```

ksc-odoo commented on content/applications/essentials/search.rst:
```diff
@@ -1,100 +1,214 @@
+
+.. example::
+   Instead of adding a :ref:`custom filter <search/custom-filters>` to select records where
+   *Mitchell Admin* is the salesperson on the *Sales Analysis* report (:menuselection:`Sales -->
```
```suggestion
   *Mitchell Admin* is the salesperson on the *Sales Analysis* report (:menuselection:`Sales app -->
```

ksc-odoo commented on content/applications/essentials/search.rst:
```diff
@@ -1,100 +1,214 @@
+.. example::
+   Instead of adding a :ref:`custom filter <search/custom-filters>` to select records where
+   *Mitchell Admin* is the salesperson on the *Sales Analysis* report (:menuselection:`Sales -->
+   Reporting --> Sales`), search for `Mitch` and click the :guilabel:`⏵ (right arrow)` next to
```
```suggestion
   Reporting --> Sales`), search for `Mitch`, and click the :guilabel:`⏵ (right arrow)` next to
```

ksc-odoo commented on content/applications/essentials/search.rst:
```diff
@@ -1,100 +1,214 @@
+
+.. note::
+   Using the search field is equivalent to using the *contains* operator when adding a :ref:`custom
+   filter <search/custom-filters>`. If a partial value is entered and the desired field is directly
```
```suggestion
   filter <search/custom-filters>`. If a partial value is entered, and the desired field is directly
```

ksc-odoo commented on content/applications/essentials/search.rst:
```diff
@@ -1,100 +1,214 @@
+   Using the search field is equivalent to using the *contains* operator when adding a :ref:`custom
+   filter <search/custom-filters>`. If a partial value is entered and the desired field is directly
+   selected (without selecting the :guilabel:`⏵ (right arrow)`), *all* records containing the typed
+   characters for the selected field will be included.
```
```suggestion
   characters for the selected field are included.
```

ksc-odoo commented on content/applications/essentials/search.rst:
```diff
@@ -1,100 +1,214 @@
+
+Filters are used to select records that meet specific criteria. The default selection of records is
+specific to each view, but can be modified by selecting one or several :ref:`preconfigured filters
+<search/preconfigured-filters>` or by adding a :ref:`custom filter <search/custom-filters>`.
```
```suggestion
<search/preconfigured-filters>`, or by adding a :ref:`custom filter <search/custom-filters>`.
```

ksc-odoo commented on content/applications/essentials/search.rst:
```diff
@@ -1,100 +1,214 @@
-You can modify the default selection of records by clicking :guilabel:`Filters` and selecting one or
-several **preconfigured filters**.
+Modify the default selection of records by clicking the :guilabel:`🔽 (down arrow) icon` from the
+search bar and selecting one or several *preconfigured filters* from the :guilabel:`Filters`
```
```suggestion
search bar, and selecting one (or several) *preconfigured filters* from the :guilabel:`Filters`
```

ksc-odoo commented on content/applications/essentials/search.rst:
```diff
@@ -1,100 +1,214 @@
+=======
+
+Filters are used to select records that meet specific criteria. The default selection of records is
+specific to each view, but can be modified by selecting one or several :ref:`preconfigured filters
```
```suggestion
specific to each view, but can be modified by selecting one (or several) :ref:`preconfigured filters
```

ksc-odoo commented on content/applications/essentials/search.rst:
```diff
@@ -1,100 +1,214 @@
-   :guilabel:`Quotations`. Furthermore, you could *only* include records from a specific year, for
-   example *2022*, by selecting :menuselection:`Order Date --> 2022`.
+   On the *Sales Analysis* report (:menuselection:`Sales --> Reporting --> Sales`), only records
+   that are at the *sales order* stage with an *order date* within the last 365 days are selected by
```
```suggestion
   that are at the *sales order* stage, with an *order date* within the last 365 days, are selected by
```

ksc-odoo commented on content/applications/essentials/search.rst:
```diff
@@ -1,100 +1,214 @@
+   that are at the *sales order* stage with an *order date* within the last 365 days are selected by
+   default.
+
+   To *also* include records at the *quotation* stage, select :guilabel:`Quotations` from the
```
```suggestion
   To also include records at the *quotation* stage, select :guilabel:`Quotations` from the
```

ksc-odoo reviewed (approved):
Hey @samueljlieber -- just finished my Final Review. Great work! Only have some super-minor adjustments that require your attention. Other than that, this looks "good to go" 👍

samueljlieber commented:
Thank you for your helpful review @ksc-odoo, I implemented your suggestions in ce0ff4e :)

samueljlieber commented:
@robodoo r+
